### ISSUE TITLE
Event with no root object.

### ISSUE THREAD

**Issue by jurek7** [wso2-extensions/siddhi-map-json#36]
I'm getting the following error:

```
[2018-01-16 16:02:02,265] ERROR {org.wso2.extension.siddhi.map.json.sourcemapper.JsonSourceMapper} - Default json message {"eventName":"START_NODE","eventCode":"2"} contains an invalid event identifier. Required "event", but found "eventName". Hence dropping the message.
```

It seems that JSON mapper doesn't support json objects with no root element. I tried to experiment with different configuration of enclosing.element parameter with no result.

Whole Siddhi script:
```
@App:name("CargoWeightApp")

@sink(type='log', prefix='LOGGER')
define stream OutputStream(eventName string);

@source(type ='rabbitmq',
    uri = 'amqp://guest:guest@localhost:5672',
    exchange.name = 'device-event',
    routing.key= '#' , @map(type='json', fail.on.missing.attribute='false')
)
define stream MetricsStream(eventName string);

@info(name='q1')
from MetricsStream
select eventName
insert into OutputStream;
```

**Comment by jurek7** [comment]
I found that this setting works as expected(which is not so obvious):
@map(type='json', enclosing.element='$', @attributes(symbol = "symbol", price = "price", volume = "volume")

enclosing element must be set to "$" and @attributes parameter **must** be provided even if custom mapping is not required.

**Comment by minudika** [comment]
We have resolved this with #47  and released v4.0.24.

**Comment by NisalaNiroshana** [comment]
Fixed with above PR
